static/js/components/LimitSelector.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ChangeEvent, Dispatch, FC, SetStateAction } from "react";
+import { Col, Input, RadioInput, Row } from "@canonical/react-components";
+
+interface Props {
+  cpuLimit: string;
+  setCpuLimit: Dispatch<SetStateAction<string>>;
+  memoryLimit: string;
+  setMemoryLimit: Dispatch<SetStateAction<string>>;
+  measure: string;
+  setMeasure: Dispatch<SetStateAction<string>>;
+}
+
+const MIB = "MiB";
+const GIB = "GiB";
+
+const LimitSelector: FC<Props> = ({
+  cpuLimit,
+  setCpuLimit,
+  memoryLimit,
+  setMemoryLimit,
+  measure,
+  setMeasure,
+}) => {
+  const updateCpuLimit = (v: ChangeEvent<HTMLInputElement>) => {
+    setCpuLimit(v.target.value);
+  };
+
+  const updateMemoryLimit = (v: ChangeEvent<HTMLInputElement>) => {
+    setMemoryLimit(v.target.value);
+  };
+
+  return (
+    <>
+      <Input
+        id="cpuLimit"
+        name="cpuLimit"
+        type="number"
+        label="CPU limit"
+        onChange={updateCpuLimit}
+        value={cpuLimit}
+        stacked
+      />
+      <Input
+        id="memoryLimit"
+        name="memoryLimit"
+        type="number"
+        label="Memory limit"
+        onChange={updateMemoryLimit}
```

Comment:
I think we should simplify this so the parent always receives the memory limit in MiB, and passes in the value as MiB. Only in here we have the additional unit selector. On change and when receiving the limit from the parent we convert to MiB. Hence the parent has not to deal with the unit complexity, we rather keep it local.